### PR TITLE
Fix an NPE in overridden_fields

### PR DESCRIPTION
**Pull request by bwilkerson**
Fixes https://github.com/dart-lang/sdk/issues/34571

### CODE REVIEW

File: lib/src/rules/overridden_fields.dart [hunk 1]
```diff
@@ -75,7 +75,11 @@ Iterable<InterfaceType> _findAllSupertypesAndMixins(
 
   accumulator.add(interface);
   InterfaceType superclass = interface.superclass;
-  Iterable<InterfaceType> interfaces = [superclass]
+  List<InterfaceType> interfaces = [];
+  if (superclass != null) {
+    interfaces.add(superclass);
```
Comment (scheglov) [inline lib/src/rules/overridden_fields.dart]: Hm... This whole function looks like a copy of `ClassElement.allSupertypes`. This method receives `InterfaceType`, but its client actually uses `classElement.type`, and actually does not care about types - just about presence of an accessor in a supertype.

File: test/rules/overridden_fields.dart [hunk 2]
```diff
@@ -162,3 +162,7 @@ class StaticsNo {
 class VerifyStatic extends StaticsNo {
   static int a;
 }
+
+mixin M on A1 {
```
Comment (scheglov) [inline test/rules/overridden_fields.dart]: Maybe also positive / negative tests for the lint?

File: lib/src/rules/overridden_fields.dart (hunk 1 above)
Comment (bwilkerson) [reply lib/src/rules/overridden_fields.dart]: Nope. This excludes implemented types, looking only along the superclass chain.

File: test/rules/overridden_fields.dart (hunk 2 above)
Comment (bwilkerson) [reply test/rules/overridden_fields.dart]: Done

**Comment by bwilkerson** [comment]
Adding support so that the rule correctly reports overrides in mixins required adding enough code that you should probably look at it again.

**Pull request merged**